### PR TITLE
Port `script` program from NetBSD

### PR DESCRIPTION
**Pull request by j-piecuch**
The `script` program records the output of a shell session to a file for later viewing.

Summary of all changes to the original program I recall having made:
- Replaced `_PATH_BSHELL` with `_PATH_KSHELL` in `doshell()`

### CODE REVIEW

**Review by bkjg — APPROVED** [review]
LGTM

File: usr.bin/script/script.c [hunk 1]
```diff
@@ -0,0 +1,481 @@
...
+	sigsuspend(&set);
+}
+
+static void
+finish(int signo)
+{
+	int die, pid, status;
+
+	die = 0;
+	while ((pid = wait4(-1, &status, WNOHANG, 0)) > 0)
```
Comment (cahirwpz) [inline usr.bin/script/script.c:218]: Could we keep original `wait3` and introduce a wrapper for `wait4` in `libc`?

File: usr.bin/script/script.c [hunk 2]
```diff
@@ -0,0 +1,481 @@
...
+			err(EXIT_FAILURE, "openpty");
+	} else {
+		if (openpty(&master, &slave, NULL, &tt, &win) == -1)
+			err(EXIT_FAILURE, "openpty");
+		isterm = 1;
+	}
+
+  if (!quiet){
+    (void)printf("Script started, output file is %s\n", fname);
+    fflush(stdout);
```
Comment (cahirwpz) [inline usr.bin/script/script.c:162]: Is it really required to introduce `fflush` here?

File: usr.bin/script/script.c [hunk 3]
```diff
@@ -0,0 +1,481 @@
...
+	struct stamp stamp;
+	struct timeval tv;
+  int fd = fileno(fp);
+
+	(void)gettimeofday(&tv, NULL);
+	stamp.scr_len = cc;
+	stamp.scr_sec = tv.tv_sec;
+	stamp.scr_usec = tv.tv_usec;
+	stamp.scr_direction = direction;
+	if (write(fd, &stamp, sizeof(stamp)) != sizeof(stamp) ||
```
Comment (cahirwpz) [inline usr.bin/script/script.c:339]: Could we introduce `writev` wrapper in `libc` or better implement the syscall ?

**Review by cahirwpz — COMMENTED** [review]
If you introduce changes into external files please adhere to their formatting.

File: usr.bin/script/script.c (hunk 1 above)
Comment (j-piecuch) [reply usr.bin/script/script.c]: Done.

File: usr.bin/script/script.c (hunk 2 above)
Comment (j-piecuch) [reply usr.bin/script/script.c]: It was necessary a couple of weeks ago, but it no longer is. I've removed it.

File: usr.bin/script/script.c (hunk 3 above)
Comment (j-piecuch) [reply usr.bin/script/script.c]: I have added userspace implementations of `writev()` and `readv()`.

**Review by cahirwpz — APPROVED** [review]
LGTM I'll merge it after #997

**Pull request merged**